mantidproject/mantid
# Remove pre-commit from CPPCheck, Doxygen, & Unit Test Builds

**Description of work.**
Disable pre-commit on CppCheck builds. There's something slightly up with the gitconfig at the moment, which was causing this issue:
```
16:30:59 [ERROR] Cowardly refusing to install hooks with `core.hooksPath` set.
16:30:59 hint: `git config --unset-all core.hooksPath`
16:30:59 CMake Error at buildconfig/CMake/CommonSetup.cmake:420 (message):
16:30:59   Pre-commit tried to install itself into your repository, but failed to do
16:30:59   so.  Is it installed on your system?
```
We don't need the pre-commit hooks in our CppCheck builds, so disable it.   

**Jenkins Changes:**
- Added the `-DENABLE_PRECOMMIT=OFF` to the Doxygen build job. 

**To test:**
1. Branch off of this PRs branch and make a PR that edits some C++. 
2. Check that CppCheck runs and any errors are only related to the changes you made. 


Fixes #34682 


*This does not require release notes* because **it's DevOps work and so isn't user facing.**




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
